src/router/controllers_v1/PlanStatusController.js

```diff
@@ -60,10 +61,39 @@ export default class PlanStatusController {
       const updatedPlan = await Plan.update(db, { id: planId }, body);
 
       // If the new status was legal, create a snapshot after updating
-      if (Plan.isLegal(updatedPlan)) {
+      if (Plan.isLegal(updatedPlan) || status.code === PLAN_STATUS.WRONGLY_MADE_WITHOUT_EFFECT) {
         await Plan.createSnapshot(db, planId, user.id);
       }
 
+      if (status.code === PLAN_STATUS.WRONGLY_MADE_WITHOUT_EFFECT) {
+        /*
+         *  The previous legal version should always be right before the "stands"
+         *  version for the minor amendment that was just determined to be
+         *  wrongly made without effect.
+         */
+        // eslint-disable-next-line no-unused-vars
+        const { rows: [stands, prevLegal] } = await db.raw(`
+          SELECT * FROM plan_snapshot_summary
+          WHERE effective_legal_end IS NOT NULL
+          AND plan_id = ?
+          ORDER BY created_at DESC;
+        `, [planId]);
+
+        const { rows: [{ max: lastVersion }] } = await db.raw(`
```

Comment:
The `lastVersion` here is actually the same as `max`. I just change the variable name in the destructuring to `lastVersion` because it adds a bit more context.